torch/nn/modules/linear.py

```diff
@@ -49,6 +49,10 @@ class Linear(Module):
         - Output: :math:`(N, *, H_{out})` where all but the last dimension
           are the same shape as the input and :math:`H_{out} = \text{out\_features}`.
 
+    .. note::
```

Comment:
wdyt about just changing the notation above to `(*, H{in})` and `(*, H_{out})` instead of adding this note?

`where * represents any number of dimensions, including none`

Reply:
I am concerned that `none` can get confused with size 0, i.e. `(0, 10, 20)`. I think having `*` mean batch + feature dimensions inflates the meaning of `*` a little too much. The explicit `N` is very good for beginners and the common use case of having batches.

If we are avoiding the note, maybe:

> `(N, *, H_{in})` where `*` represents ... . The 0th dimension is optional.

Reply:
I was going to suggest linking to a [glossary](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-glossary) that defines "no batch dimension" but there is not a glossary in the pytorch docs. (yet)